### PR TITLE
[code-completion] Fix crash in printParameterList when paramListTy is…

### PR DESCRIPTION
**Pull request by nathawes**
This fixes a crash exposed by validation-test/IDE/crashers/101-swift-decl-print.swift.  When paramListTy was Error type, it was still treated as a valid parameter list type. This resulted in a path through the method where the paramFlags vector had fewer elements than the parameter list, PL, so paramFlags[i], where i is an index into PL, triggered an assert.

Resolves rdar://problem/28867801.

… Error type rdar://problem/28867801

### CODE REVIEW

**Comment by nathawes** [comment]
@swift-ci please smoke test

File: lib/AST/ASTPrinter.cpp [hunk 1]
```diff
@@ -2461,7 +2461,7 @@ void PrintAST::printParameterList(ParameterList *PL, Type paramListTy,
                                   bool isCurried,
                                   std::function<bool()> isAPINameByDefault) {
   SmallVector<ParameterTypeFlags, 4> paramFlags;
-  if (paramListTy) {
+  if (paramListTy && !paramListTy.getPointer()->hasError()) {
```
Comment (jrose-apple) [inline lib/AST/ASTPrinter.cpp]: Type has an overloaded `operator->`, so you can just use `paramListTy->hasError()`.

Comment (nathawes) [inline lib/AST/ASTPrinter.cpp]: Thanks, I'll fix it up!

**Comment by nathawes** [comment]
@swift-ci please smoke test

**Comment by nathawes** [comment]
@swift-ci please smoke test

**Pull request merged**